pkg/kubelet/kubelet.go

```diff
@@ -2962,29 +2977,20 @@ func (kl *Kubelet) handlePodResourcesResize(pod *v1.Pod, podStatus *kubecontaine
 				}
 			}
 		}
-		allocatedPod = pod
 
-		// Special case when the updated allocation matches the actuated resources. This can occur
+		// Special case when the updated allocation matches the actuated resources, which can occur
 		// when reverting a resize that hasn't been actuated, or when making an equivalent change
-		// (such as CPU requests below MinShares). This is an optimization to clear the resize
-		// status immediately, rather than waiting for the next SyncPod iteration.
-		if !kl.isPodResizeInProgress(allocatedPod, podStatus) {
-			// In this case, consider the resize complete.
-			kl.statusManager.SetPodResizeStatus(pod.UID, "")
-			return allocatedPod, nil
-		}
-	}
-	if resizeStatus != "" {
-		kl.statusManager.SetPodResizeStatus(pod.UID, resizeStatus)
-		if resizeMsg != "" {
-			switch resizeStatus {
-			case v1.PodResizeStatusDeferred:
-				kl.recorder.Eventf(pod, v1.EventTypeWarning, events.ResizeDeferred, resizeMsg)
-			case v1.PodResizeStatusInfeasible:
-				kl.recorder.Eventf(pod, v1.EventTypeWarning, events.ResizeInfeasible, resizeMsg)
-			}
-		}
+		// (such as CPU requests below MinShares). We only want to clear the PodResizeInProgress condition
+		// if allocation matches actuation. If they do not match, we skip setting the PodResizeInProgress
+		// condition and let the next SyncPod iteration handle it.
+		setResizeInProgressCondition = false
```

Comment:
I’m not familiar enough (yet) with this feature’s implementation to tell you why, but this line is needed to prevent [these test failures](https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/130733/pull-kubernetes-e2e-inplace-pod-resize-containerd-main-v2/1900673211519995904). I noticed in the existing implementation, the [isPodResizeInProgress check](https://github.com/kubernetes/kubernetes/blob/157f42bff333811956531ea692cc1fdfa0d6a0a4/pkg/kubelet/kubelet.go#L2971) is used only to clear the resize status, not to set it. Seems if we set it here, it doesn’t always get cleared at the right time, so better to let the next SyncPod iteration handle setting it.